misc/package/Data Files/MWSE/core/mwse/travelFinished/main.lua

```diff
@@ -0,0 +1,34 @@
+local timeBeforeTravel = 0
+local payload = {}
+
+---@param e cellChangedEventData
+local function travelEnded(e)
+	payload.cell = e.cell
+	payload.previousCell = e.previousCell
+	event.trigger(tes3.event.travelFinished, payload)
+	event.unregister(tes3.event.cellChanged, travelEnded)
+end
+
+---@param e calcTravelPriceEventData
+local function travelStart(e)
+	if (not tes3.mobilePlayer.traveling) then -- Get time before traveling
+		timeBeforeTravel = tes3.getSimulationTimestamp()
+	end
+
+	if tes3.mobilePlayer.traveling then	-- Travel finished, but calcTravelPrice triggered once again.
+		table.copy(e, payload)
+		payload.claim = nil
+		payload.hoursPassed = tes3.getSimulationTimestamp() - timeBeforeTravel
+		timeBeforeTravel = 0
+
+		-- calcTravelPrice event is triggered once for every possible destination on
+		-- first opening of the Travel Menu. Once again after a destination has been
+		-- selected, but before the player was teleported. This event is triggered
+		-- after the player is teleported to the destination.
+		event.register(tes3.event.cellChanged, travelEnded)
+	end
+end
+
+event.register(tes3.event.initialized, function()
+	event.register(tes3.event.calcTravelPrice, travelStart, { priority = -100 })
+end)
```

Comment:
This event could be implemented a lot more straightforward in C, but I dont have an issue with it being lua only. This doesnt look like it will impact performance in the slightest. Maybe @NullCascade could chime in on that.

Reply:
If anybody is interested to rewrite this in C, feel free to. This file can easily be removed at that point.